src/odemis/driver/smaract.py

```diff
@@ -3173,3 +3174,1239 @@ def SA_CTL_Move(self, handle, ch, pos_pm, _):
+        property_key (int64): property key symbol
+        value (int64): value to set
+        idx0 (int): index value, meaning depends on property key
+        idx1 (int): index value, meaning depends on property key
+        """
+        encoded_key = self.core.SA_SI_EPK(property_key, idx0, idx1)
+        self.core.SA_SI_SetProperty_i64(self._id, c_uint32(encoded_key), c_int64(value))
+
+    def GetProperty_f64(self, property_key, idx0=0, idx1=0):
+        """
+        property_key (int32): property key symbol
+        idx0 (int): index value, meaning depends on property key
+        idx1 (int): index value, meaning depends on property key
+        returns (float) the value
+        """
+        ret_val = c_double()
+        encoded_key = self.core.SA_SI_EPK(property_key, idx0, idx1)
+        self.core.SA_SI_GetProperty_f64(self._id, c_uint32(encoded_key), byref(ret_val), c_size_t(0))
+        return ret_val.value
+
+    def GetProperty_i32(self, property_key, idx0=0, idx1=0):
+        """
+        property_key (int32): property key symbol
+        idx0 (int): index value, meaning depends on property key
+        idx1 (int): index value, meaning depends on property key
+        returns (int) the value
+        """
+        ret_val = c_int32()
+        encoded_key = self.core.SA_SI_EPK(property_key, idx0, idx1)
+        self.core.SA_SI_GetProperty_i32(self._id, c_uint32(encoded_key), byref(ret_val), c_size_t(0))
+        return ret_val.value
+
+    def GetProperty_i64(self, property_key, idx0=0, idx1=0):
+        """
+        property_key (int64): property key symbol
+        idx0 (int): index value, meaning depends on property key
+        idx1 (int): index value, meaning depends on property key
+        returns (int) the value
+        """
+        ret_val = c_int64()
+        encoded_key = self.core.SA_SI_EPK(property_key, idx0, idx1)
+        self.core.SA_SI_GetProperty_i64(self._id, c_uint32(encoded_key), byref(ret_val), c_size_t(0))
+        return ret_val.value
+
+    def GetProperty_s(self, property_key, idx0=0, idx1=0):
+        """
+        property_key (int32): property key symbol
+        idx0 (int): index value, meaning depends on property key
+        idx1 (int): index value, meaning depends on property key
+        returns (str): the value
+        """
+        ret_val = create_string_buffer(SA_SI_STRING_MAX_LENGTH)
+        encoded_key = self.core.SA_SI_EPK(property_key, idx0, idx1)
+        self.core.SA_SI_GetProperty_s(self._id, c_uint32(encoded_key), ret_val, byref(c_size_t(len(ret_val))))
+        return ret_val.value.decode("latin1")
+
+    def GetValue_f64(self, channel, data_source_idx):
+        """
+        property_key (int32): property key symbol
+        data_source_idx:
+        returns (float) the value
+        """
+        ret_val = c_double()
+        self.core.SA_SI_GetValue_f64(self._id, channel, data_source_idx, byref(ret_val))
+        return ret_val.value
+
+    def WaitForEvent(self, timeout=600):
```

Comment:
600 as a default is odd. I'd say either don't put any default, or put infinity.

Reply:
I really don't understand your previous comment then. You told me to **not** put infinity, but a value like 600. What do you mean with that?

Reply:
In the function *signatures*, ie, here and line 3953, you set the default to infinity (or no default).
In the function *callers*, l 4018, 4037, 4083, you explicitly indicate a timeout. That timeout could be around 10 min, or less now that we saw a properly configured hardware is faster.